homeassistant/components/mqtt/client.py

```diff
@@ -295,6 +297,81 @@ def _is_simple_match(topic: str) -> bool:
     return not ("+" in topic or "#" in topic)
 
 
+class EnsureJobAfterCooldown:
```

Comment:
Isn't this just re-implementing the debouncer that exists in HASS?

Reply:
We restart the debouncer timer if a new subscribtion comes in.

Reply:
Wouldn't you then want some kind of maximum time?

If you cancel/restart every time a subscribe comes in, you could delay for a LONG time.  eg. if I kept sending in subscribes, it could easily delay any subscriptions for minutes because of timing.

Not quite as likely, but we should protect against starvation/infinite delays.